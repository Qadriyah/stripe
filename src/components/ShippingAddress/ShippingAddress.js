import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Notifications from 'react-notify-toast';
import { addCourse } from '../../actions/courseAction';
import {
  renderTextInput,
  renderButton,
  renderToggleButton
} from '../TextInputs/inputHelpers';
import { showNotification, getFormData } from '../TextInputs/helper';
import './ShippingAddress.scss';

class ShippingAddress extends Component {
  constructor(props) {
    super(props);
    this.state = {
      inputs: {}
    };
  }

  onSubmit = event => {
    event.preventDefault();
    const { inputs } = this.state;
    const inputData = getFormData(inputs);
    let message = 'Data submitted successfully';
    let className = 'toast__message toast__message_success';
    if (!inputData.isValid) {
      message = 'All fields are required';
      className = 'toast__message toast__message_error';
    }
    showNotification({
      message,
      className,
      type: 'custom'
    });
  };

  renderForm = () => (
    <form onSubmit={this.onSubmit} noValidate>
      <div className="row">
        {renderTextInput({
          name: 'name',
          placeholder: 'First and last name',
          label: 'Name',
          className: 'form-control form-control-lg',
          serverError: '',
          parent: this,
          col: 'col-12',
          required: true
        })}
        {renderTextInput({
          name: 'address',
          placeholder: 'Street address',
          label: 'Address',
          className: 'form-control form-control-lg',
          serverError: '',
          parent: this,
          col: 'col-12',
          required: true
        })}
        {renderTextInput({
          name: 'address2',
          placeholder: 'Street address 2 (Optional)',
          className: 'form-control form-control-lg',
          serverError: '',
          parent: this,
          col: 'col-12'
        })}
        {renderTextInput({
          name: 'city',
          placeholder: 'City',
          className: 'form-control form-control-lg',
          serverError: '',
          parent: this,
          col: 'col-7',
          required: true
        })}
        {renderTextInput({
          name: 'state',
          placeholder: 'State',
          className: 'form-control form-control-lg',
          serverError: '',
          parent: this,
          col: 'col-5',
          required: true
        })}
        {renderTextInput({
          name: 'zip',
          placeholder: 'Zip code',
          className: 'form-control form-control-lg',
          serverError: '',
          parent: this,
          col: 'col-5',
          required: true
        })}
      </div>
      <div className="row">
        {renderButton({
          type: 'submit',
          className: 'btn btn-primary btn-lg btn-block',
          col: 'col-12'
        })}
      </div>
    </form>
  );

  render() {
    const { status } = this.props;
    const styles = status ? 'grade-form__body' : 'grade-form__body-1';
    return (
      <>
        <div className="grade-form mb-3">
          <Notifications options={{ zIndex: 200, top: '5px' }} />
          <div className="grade-form__header">
            <h5>Shipping to</h5>
          </div>
          <div className={styles}>
            {status ? this.renderForm() : renderToggleButton(this)}
          </div>
        </div>
      </>
    );
  }
}

ShippingAddress.propTypes = {
  status: PropTypes.bool.isRequired
};

export default connect(
  null,
  { addCourse }
)(ShippingAddress);
