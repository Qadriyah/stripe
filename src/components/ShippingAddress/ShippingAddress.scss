@import '../../assets/abstract/variables';

.grade-form {
  box-sizing: border-box;
  border: 1px solid $border-color;
  border-radius: 10px;
  background-color: #ffffff;

  &__header {
    background-color: $background-color;
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
    padding: 20px;
    border-bottom: solid thin $border-color;
  }

  &__body {
    padding: 20px;
  }

  &__body-1 {
    padding: 20px;
    background-color: $background-color;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
  }
}
