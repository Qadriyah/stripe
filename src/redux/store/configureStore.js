import { applyMiddleware, createStore } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers';

const configureStore = preloadedState => {
  const loggerMiddleware = createLogger();
  const middlewares = [thunkMiddleware];

  if (process.env.NODE_ENV !== 'production') middlewares.push(loggerMiddleware);
  const middlewareEnhancers = applyMiddleware(...middlewares);

  const enhancers = [middlewareEnhancers];
  const composedEnhancers = composeWithDevTools(...enhancers);

  const store = createStore(rootReducer, preloadedState, composedEnhancers);

  if (process.env.NODE_ENV !== 'production' && module.hot)
    module.hot.accept('../reducers', () => store.replaceReducer(rootReducer));

  return store;
};

export default configureStore;
