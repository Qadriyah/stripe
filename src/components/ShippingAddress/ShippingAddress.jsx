import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { processFormData } from '../TextInputs/helper';
import { isEmpty } from '../../utils';
import './ShippingAddress.scss';

class ShippingAddress extends Component {
  state = {
    inputs: {},
    message: '',
    className: ''
  };

  handleSubmit = event => {
    event.preventDefault();
    const { inputs } = this.state;
    const inputData = processFormData(inputs);
    let message = 'Data submitted successfully';
    let className = 'toast__message toast__message_success';
    if (!inputData.isValid) {
      message = 'All fields are required';
      className = 'toast__message toast__message_error';
    }
    this.setState({
      message,
      className
    });
  };

  renderForm = () => {
    const { renderTextInput, renderButton } = this.props;
    return (
      <form noValidate>
        <div className="row">
          {renderTextInput({
            name: 'name',
            placeholder: 'First and last name',
            label: 'Name',
            parent: this,
            col: 'col-12',
            required: true
          })}
          {renderTextInput({
            name: 'address',
            placeholder: 'Street address',
            label: 'Address',
            parent: this,
            col: 'col-12',
            required: true
          })}
          {renderTextInput({
            name: 'address2',
            placeholder: 'Street address 2 (Optional)',
            parent: this,
            col: 'col-12'
          })}
          {renderTextInput({
            name: 'city',
            placeholder: 'City',
            parent: this,
            col: 'col-7',
            required: true
          })}
          {renderTextInput({
            name: 'state',
            placeholder: 'State',
            parent: this,
            col: 'col-5',
            required: true
          })}
          {renderTextInput({
            name: 'zip',
            placeholder: 'Zip code',
            parent: this,
            col: 'col-5',
            required: true
          })}
        </div>
        <div className="row">
          {renderButton({
            type: 'submit',
            col: 'col-5',
            value: 'Save',
            onClick: this.handleSubmit
          })}
        </div>
      </form>
    );
  };

  renderFormBody = ({ status, renderButton, handleButtonClick }) => {
    return status ? (
      this.renderForm()
    ) : (
      <div className="row">
        {renderButton({
          className: 'btn btn-general btn-lg btn-block',
          col: 'col-5',
          value: 'Add credit card',
          onClick: handleButtonClick
        })}
      </div>
    );
  };

  render() {
    const { status } = this.props;
    const { message, className } = this.state;
    const styles = status ? 'grade-form__body' : 'grade-form__body-1';
    return (
      <>
        <div className="grade-form mb-3">
          <div className="grade-form__header">
            <h5> Shipping to </h5>
          </div>
          <div className={styles}>{this.renderFormBody(this.props)}</div>
          {!isEmpty(message) && <div className={className}> {message} </div>}
        </div>
      </>
    );
  }
}

ShippingAddress.propTypes = {
  status: PropTypes.bool.isRequired,
  renderButton: PropTypes.func.isRequired,
  renderTextInput: PropTypes.func.isRequired
};

export default ShippingAddress;
