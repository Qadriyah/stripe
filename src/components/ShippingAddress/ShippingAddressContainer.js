import { connect } from 'react-redux';
import { renderTextInput, renderButton } from '../TextInputs/inputHelpers';
import ShippingAddress from './ShippingAddress';

export default connect(
  null,
  {
    renderTextInput,
    renderButton
  }
)(ShippingAddress);
