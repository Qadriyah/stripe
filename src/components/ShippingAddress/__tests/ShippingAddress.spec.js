import React from 'react';
import { mount, shallow } from 'enzyme';
import Thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import ShippingAddress from '../ShippingAddressContainer';

jest.useFakeTimers();

describe('PaymentMethod', () => {
  const mockStore = configureStore([Thunk]);
  const store = mockStore({});
  const defaultProps = {
    renderTextInput: jest.fn(),
    renderButton: jest.fn(),
    handleButtonClick: jest.fn(),
    status: true
  };

  const address = {
    name: 'Baker Sekitoleko',
    address: 'Entebbe road',
    city: 'Kampala',
    state: 'Najjanankumbi',
    zip: '00256'
  };

  const setup = (newProps, shouldMount = false) => {
    const props = { ...defaultProps, ...newProps };
    const wrapper = shouldMount
      ? mount(
          <Provider store={store}>
            <ShippingAddress {...props} />
          </Provider>
        )
      : shallow(
          <Provider store={store}>
            <ShippingAddress {...props} />
          </Provider>
        );
    return wrapper;
  };

  const addInputValue = (wrapper, input, value) =>
    wrapper.find(input).simulate('change', { target: { value } });

  test('it renders without errors', () => {
    const wrapper = setup({});
    expect(wrapper).toMatchSnapshot();
  });

  test('it should submit the form successfully', () => {
    const wrapper = setup({}, true);

    addInputValue(wrapper, '#name', address.name);
    addInputValue(wrapper, '#address', address.address);
    addInputValue(wrapper, '#city', address.city);
    addInputValue(wrapper, '#state', address.state);
    addInputValue(wrapper, '#zip', address.zip);

    const button = wrapper.find("[data-test='component-button-submit']");
    button.simulate('click');

    const component = wrapper.find('ShippingAddress').instance();
    expect(component.state.message.trim().length).toBeGreaterThan(0);
  });

  test('it should submit the form with errors', () => {
    const wrapper = setup({}, true);

    addInputValue(wrapper, '#name', address.name);
    addInputValue(wrapper, '#address', address.address);

    const button = wrapper.find("[data-test='component-button-submit']");
    button.simulate('click');

    const component = wrapper.find('ShippingAddress').instance();
    expect(component.state.message.trim().length).toBeGreaterThan(0);
  });

  test('it should render the toggle button', () => {
    const wrapper = setup({ status: false }, true);
    expect(wrapper).toMatchSnapshot();
  });

  test('componentDidUpdate method', () => {
    jest.advanceTimersByTime(5000);
    const wrapper = setup({ status: false }, true);
    const component = wrapper.find('ShippingAddress').instance();
    component.componentDidUpdate();
    expect(setTimeout).toHaveBeenCalled();
    expect(component.state.message.trim()).toEqual('');
  });
});
