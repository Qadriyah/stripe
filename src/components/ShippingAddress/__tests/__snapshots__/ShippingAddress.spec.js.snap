// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentMethod it renders without errors 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <React.Memo
    handleButtonClick={[MockFunction]}
    renderButton={[MockFunction]}
    renderTextInput={[MockFunction]}
    status={true}
  />
</ContextProvider>
`;

exports[`PaymentMethod it should render the toggle button 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ShippingAddress)
    handleButtonClick={[MockFunction]}
    renderButton={[MockFunction]}
    renderTextInput={[MockFunction]}
    status={false}
  >
    <ShippingAddress
      handleButtonClick={[MockFunction]}
      renderButton={[Function]}
      renderTextInput={[Function]}
      status={false}
    >
      <div
        className="grade-form mb-3"
      >
        <div
          className="grade-form__header"
        >
          <h5>
             Shipping to 
          </h5>
        </div>
        <div
          className="grade-form__body-1"
        >
          <div
            className="row"
          >
            <Button
              className="btn btn-general btn-lg btn-block"
              col="col-5"
              dataTest="component-button-submit"
              onClick={[MockFunction]}
              type="button"
              value="Add credit card"
            >
              <div
                className="col-5"
              >
                <input
                  className="btn btn-general btn-lg btn-block"
                  data-test="component-button-submit"
                  onClick={[MockFunction]}
                  type="button"
                  value="Add credit card"
                />
              </div>
            </Button>
          </div>
        </div>
      </div>
    </ShippingAddress>
  </Connect(ShippingAddress)>
</Provider>
`;
